akka-stream-tests/src/test/scala/akka/stream/scaladsl/LazySinkSpec.scala

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import java.util.concurrent.TimeoutException
+
+import akka.stream.ActorAttributes.supervisionStrategy
+import akka.stream.Supervision._
+import akka.stream._
+import akka.stream.testkit.TestPublisher
+import akka.stream.testkit.TestSubscriber.Probe
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.scaladsl.TestSink
+import akka.testkit.AkkaSpec
+
+import scala.concurrent.{ Promise, Future, Await }
+import scala.concurrent.duration._
+
+class LazySinkSpec extends AkkaSpec {
+
+  val settings = ActorMaterializerSettings(system)
+    .withInputBuffer(initialSize = 1, maxSize = 1)
+  implicit val materializer = ActorMaterializer(settings)
+
+  val fallback = () ⇒ fail("Must not call fallback function")
+  val ex = TE("")
+
+  "A LazySink" must {
+    "work in happy case" in assertAllStagesStopped {
+      val futureProbe = Source(0 to 10).runWith(Sink.`lazy`[Int, Probe[Int]](_ ⇒ Future.successful(TestSink.probe[Int]), fallback))
```

Comment:
aaargh, I missed this. We cannot use the name "lazy", I expect many developers are unaware of this notation that avoids clashes with keywords. We need to find a new name :(